File: 219-Contains-Duplicate-II.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function (nums, k) {
    map = new Map()

    for (let i = 0; i < nums.length; i++)
        if (map.has(nums[i]) && i - map.get(nums[i]) <= k)
            return true
        else
            map.set(nums[i], i)

    return false
};

File: 219-Contains-Duplicate-II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        m = {}

        for i, num in enumerate(nums):
            if num in m and abs(m[num] - i) <= k:
                return True
            else:
                m[num] = i

        return False
        
